### PR TITLE
[java] Update Gradle to 7.2

### PR DESCRIPTION
**Pull request by wing328**
Update Gradle to 7.2 (the latest stable release)

To close https://github.com/OpenAPITools/openapi-generator/issues/10517

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @nmuesch (2021/01)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Java/gradle-wrapper.properties.mustache [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-rc-3-bin.zip
```
Comment (agilob) [inline modules/openapi-generator/src/main/resources/Java/gradle-wrapper.properties.mustache:3]: Use latest stable (or LTS?) version not `rc`

File: modules/openapi-generator/src/main/resources/Java/gradlew.mustache [hunk 2]
```diff
@@ -1,7 +1,7 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
Comment (agilob) [inline modules/openapi-generator/src/main/resources/Java/gradlew.mustache:1]: is this copy-paste from another source?

File: samples/client/petstore/java/apache-httpclient/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-rc-3-bin.zip
```
Comment (agilob) [inline samples/client/petstore/java/apache-httpclient/gradle/wrapper/gradle-wrapper.properties:3]: use stable version not rc

**Comment by agilob** [comment]
`https://downloads.gradle-dn.com/distributions/gradle-7.2-bin.zip` not `https://services.gradle.org/distributions/gradle-7.2-rc-3-bin.zip` :)

**Comment by wing328** [comment]
Thanks for the review. I obtained these values/files from https://github.com/gradle/gradle/tree/v7.2.0/gradle/wrapper and https://github.com/gradle/gradle/tree/v7.2.0

**Review by agilob — APPROVED** [review]

**Pull request merged**